test/util.go

```diff
@@ -0,0 +1,30 @@
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
+func SerializeAndDeserialize(s interface{}, d string) error {
```

Comment:
Could this be converted to a test helper function? And we tend to use the testify library for assertions (https://github.com/stretchr/testify). 
```suggestion
func SerializeAndDeserialize(t *t.Testing, s interface{}, d string) {
  t.Helper()
  require.Equal(t,string(j),d)
etc ...
}
```

Reply:
will look into testify, and sure I can pass the test struct through to the function